judge/runguard.c

```diff
@@ -254,6 +263,24 @@ void error(int errnum, const char *format, ...)
 
 	write_meta("internal-error","%s: %d - %s","runguard error", errnum, format);
 
+	/* Make sure that all children are killed before terminating */
+	if ( child_pid != -1) {
```

Comment:
I think this check should be `child_pid > 0` since `child_pid == 0` when in the child fork, but there an `error()` call would work fine.

Reply:
Good point, that's right